lib/client-assets.php

```diff
@@ -695,3 +695,15 @@ function gutenberg_extend_settings_custom_units( $settings ) {
 	register_pattern( 'core/numbered-features', gutenberg_load_block_pattern( 'numbered-features' ) );
 	register_pattern( 'core/its-time', gutenberg_load_block_pattern( 'its-time' ) );
 }
+
+/**
+ * Adds the global styles support flag to the frontend.
+ */
+add_action( 'init', function() {
```

Comment:
Are we allowed to use inline functions yet or not?

Reply:
I thought it was a no for hooking them into actions/filters (because that way you can't unhooked), but it was fine for any other case?

Reply:
Also: I'd think we want to add this check within `lib/global-styles.php` instead?

Reply:
Would it be best to pass this flag as part of the block settings?

Reply:
That was my first thinking but unfortunately, we can't here because we can't rely on block settings on "save" functions as these are "pure functions", that setting should be set very early.

Reply:
To make sure I understand it: the problem is a race-condition by which block settings may be unset when the save function operates? It doesn't have anything to do with pure/impure, right? Otherwise, I don't think I have a good grasp on why using a global _from the window object_ is any different than using a global _from the block-editor data store_.